src/cipher.rs

```diff
@@ -86,6 +87,30 @@ pub extern "C" fn rustls_supported_ciphersuite_get_suite(
     supported_ciphersuite.suite.get_u16()
 }
 
+/// Write the name of the ciphersuite into the provided buffer, up to a max of
+/// `len` bytes. Output is NUL terminated. Return the number of bytes written
+/// before the NUL.
+#[no_mangle]
+pub extern "C" fn rustls_supported_ciphersuite_get_name(
+    supported_ciphersuite: *const rustls_supported_ciphersuite,
+    buf: *mut c_char,
+    len: size_t,
+) -> size_t {
+    let write_buf: &mut [u8] = unsafe {
+        if buf.is_null() {
+            return 0;
+        }
+        slice::from_raw_parts_mut(buf as *mut u8, len as usize)
+    };
+    let supported_ciphersuite = try_ref_from_ptr!(supported_ciphersuite);
+    let suite_name: String = format!("{:?}", supported_ciphersuite);
```

Comment:
Unless I'm mistaken, this will end up invoking the the `std::fmt::Debug` implementation on [`SupportedCipherSuite`](https://docs.rs/rustls/0.19.1/rustls/struct.SupportedCipherSuite.html), which includes quite a bit more than the name. That seems a little risky, since [`Debug` implementations are generally understood not to be stable](https://doc.rust-lang.org/std/fmt/trait.Debug.html). If this function is exposed just so you can get a string to print out to logs from a C program, then it's fine, but `rustls-ffi` should make it clear that the string obtained through this function should not be parsed by code because its format could change at any time. I think mentioning that in the doccomment should suffice.

Reply:
Oh, I missed that! What you're looking for is the `CipherSuite`, not the `SupportedCipherSuite`. The former should be easily accessible from the latter, though.

Reply:
True, but I think my point about `std::fmt::Debug` being unstable still applies.

Reply:
I think that's OK, for the use case that I need it for, which is compatibility with the existing psql implementation - the OpenSSL one shows the cipher you use to negotiate TLS, so I'm hoping to show something similar.

Reply:
Another nit: can we get the cipher suite name as a `&'static str` and avoid allocating a `String`?

Reply:
Not right now, but see https://github.com/rustls/rustls/pull/831.